src/third_party/dartino/two_space_heap.cc

```diff
@@ -339,6 +374,10 @@ void TwoSpaceHeap::compact_heap() {
   new_space->iterate_objects(&new_space_visitor);
 
   process_heap_->iterate_roots(&fix);
+  // At this point dead objects have been cleared out of the finalizer lists.
+  EverythingIsAlive yes;
+  process_heap_->process_registered_finalizers(&fix, &yes);
```

Comment:
We have to process the finalizers again at the end when the destination locations have been calculated by the compactor.